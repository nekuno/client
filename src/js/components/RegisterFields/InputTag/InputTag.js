import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Input from '../../ui/Input/Input.js';
import Chip from '../../ui/Chip/Chip.js';
import styles from './InputTag.scss';

export default class InputTag extends Component {

    static propTypes = {
        placeholder    : PropTypes.string,
        tags           : PropTypes.array,
        selectedLabel  : PropTypes.string,
        chipsColor     : PropTypes.oneOf(['purple', 'blue', 'pink', 'green']),
        onChangeHandler: PropTypes.func,
        onClickHandler : PropTypes.func
    };

    constructor(props) {
        super(props);

        this.handleChange = this.handleChange.bind(this);
        this.handleClick = this.handleClick.bind(this);

        this.state = {
            selected: [],
            text: ''
        }
    }

    handleChange(text) {
        if (this.props.onChangeHandler) {
            this.props.onChangeHandler(text);
        }
        this.setState({text: text});
    }

    handleClick(text) {
        const {selected} = this.state;
        let newSelected = selected.slice(0);
        const index = selected.findIndex(tag => tag === text);

        if (index !== -1) {
            newSelected.splice(index, 1);
        } else {
            newSelected.push(text);
        }
        this.setState({selected: newSelected});

        this.refs["input"].clearValue();

        if (this.props.onClickHandler) {
            this.props.onClickHandler(newSelected);
        }
    }

    render() {
        const {tags, placeholder, selectedLabel, chipsColor} = this.props;
        const {selected, text} = this.state;

        return (
            <div className={styles.inputTag}>
                <Input ref="input" placeholder={placeholder} searchIcon={true} size={'small'} onChange={this.handleChange} doNotScroll={true}/>

                {text && text.length > 1 && !selected.some(selectedTag => selectedTag === text) ?
                    <div className={styles.suggestedChip}>
                        <Chip onClickHandler={this.handleClick}
                              text={text}
                              color={chipsColor}
                        />
                    </div> : null
                }
                {tags.filter(tag => !selected.some(selectedTag => selectedTag === tag)).map((tag, index) =>
                    <div key={index} className={styles.suggestedChip}>
                        <Chip onClickHandler={this.handleClick}
                              text={tag}
                              color={chipsColor}
                        />
                    </div>
                )}
                {selected.length > 0 ?
                    <div className={styles.selectedLabel + " small"}>{selectedLabel}</div> : null
                }
                {selected.map((tag, index) =>
                    <div key={index} className={styles.selectedChip}>
                        <Chip onClickHandler={this.handleClick}
                              text={tag}
                              color={chipsColor}
                              selected={true}
                        />
                    </div>
                )}
            </div>
        );
    }
}