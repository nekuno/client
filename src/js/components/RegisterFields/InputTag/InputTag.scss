.input-tag {

  .title {
    font-weight: 600;
  }
  .suggested-chip, .selected-chip {
    display: block;
    margin: 0 16px;
    text-align: left;
  }
  .selectedLabel {
    font-weight: 600;
    margin: 8px 24px;
    text-align: left;
  }
}