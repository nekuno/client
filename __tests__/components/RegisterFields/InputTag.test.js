import React from 'react';
import InputTag from '../../../src/js/components/RegisterFields/InputTag/InputTag.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputTag component', () => {
    const mockChangeFn = jest.fn();
    const mockClickFn = jest.fn();
    const tags = ["Lorem", "Ipsum", "Lorem ipsum", "Sed ut perspiciatis", "perspiciatis"];

    const inputTag = shallow(<InputTag onClickHandler={mockClickFn} onChangeHandler={mockChangeFn} placeholder={"Foo"} tags={tags}/>);

    it('should be defined', () => {
        expect(InputTag).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputTag).toMatchSnapshot();
    });
});
