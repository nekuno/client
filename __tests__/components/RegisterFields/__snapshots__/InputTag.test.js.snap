// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputTag component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputTag
    onChangeHandler={[MockFunction]}
    onClickHandler={[MockFunction]}
    placeholder="Foo"
    tags={
      Array [
        "Lorem",
        "Ipsum",
        "Lorem ipsum",
        "Sed ut perspiciatis",
        "perspiciatis",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Input
          doNotFocus={true}
          doNotScroll={true}
          onChange={[Function]}
          placeholder="Foo"
          searchIcon={true}
          size="small"
        />,
        null,
        null,
        null,
        Array [],
      ],
      "className": "inputTag",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "doNotFocus": true,
          "doNotScroll": true,
          "onChange": [Function],
          "placeholder": "Foo",
          "searchIcon": true,
          "size": "small",
        },
        "ref": "input",
        "rendered": null,
        "type": [Function],
      },
      null,
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Input
            doNotFocus={true}
            doNotScroll={true}
            onChange={[Function]}
            placeholder="Foo"
            searchIcon={true}
            size="small"
          />,
          null,
          null,
          null,
          Array [],
        ],
        "className": "inputTag",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "doNotFocus": true,
            "doNotScroll": true,
            "onChange": [Function],
            "placeholder": "Foo",
            "searchIcon": true,
            "size": "small",
          },
          "ref": "input",
          "rendered": null,
          "type": [Function],
        },
        null,
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
